openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -272,6 +275,26 @@ def _track_user_login(user, request):
     )
 
 
+def _check_user_auth_flow(site, user):
+    """
+    Check if user blongs to an allowed domain and not whitelisted
+    then ask user to login through allowed domain SSO provider.
+    """
+    if user and waffle().is_enabled(ENABLE_LOGIN_USING_THIRDPARTY_AUTH_ONLY):
+        allowed_domain = site.configuration.get_value('THIRD_PARTY_AUTH_ONLY_DOMAIN', '').lower()
```

Comment:
If you you get rid of the waffle flag, you could just make the feature conditional on whether `THIRD_PARTY_AUTH_ONLY_DOMAIN` is set to a valid value.